src/main/java/org/overture/ego/controller/AuthController.java

```diff
@@ -92,15 +97,19 @@ boolean verifyJWToken(
   private String generateUserToken(Map authInfo){
     val userName = authInfo.get("email").toString();
     User user = userService.getByName(userName, false);
+
     if (user == null) {
       user = createNewUser(userName,authInfo);
-      userService.create(user);
+      if (saveUsersAllowed) {
+        // Saving user controlled through config - provides mechanism to run demos without saving
+        userService.create(user);
+      }
     }
     return tokenService.generateToken(user);
   }
 
   private User createNewUser(String userName, Map authInfo) {
-    String role = "USER";
+    String role = newUserRole;
```

Comment:
could we do this rather.. demo == true ? "ADMIN" : "USER"

Also; could you please create constant for these Strings. Sorry it wasn't done in previous versions.